src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,38 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int VALID_PASSWORD_LENGTH = 9;
+    private static final int VALID_USER_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null || user.getLogin().equals("")) {
+            throw new RegistrationException("User login field "
+                    + "cannot be empty");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User login "
+                    + "already exists");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("User password field "
+                    + "cannot be empty");
+        }
+        if (user.getPassword().length() < VALID_PASSWORD_LENGTH) {
+            throw new RegistrationException("User password less then "
+                    + VALID_PASSWORD_LENGTH
+                    + " symbols");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("User age field "
+                    + "cannot be empty");
+        }
+        if (user.getAge() < VALID_USER_AGE) {
+            throw new RegistrationException("Users age must be more "
+                    + VALID_USER_AGE
+                    + " years");
+        }
```

Comment:
For the future, It is better to divide into methods and describe the necessary logic in them.
For example, you can create private methods:
private checkAge(User user) {
// age verification
}